tests/integration/helpers.py

```diff
@@ -372,3 +399,121 @@ def cluster_name(unit: Unit, model_name: str) -> str:
+
+        return pid
+    except Exception:
+        return None
+
+
+@retry(stop=stop_after_attempt(12), wait=wait_fixed(15), reraise=True)
+async def is_unit_in_cluster(ops_test: OpsTest, unit_name: str, action_unit_name: str) -> boolean:
+    """Check is unit is online in the cluster.
+
+    Args:
+        ops_test: The ops test object passed into every test case
+        unit_name: The name of the unit to be tested
+        action_unit_name: a different unit to run get status action
+    Returns:
+        A boolean
+    """
+    _, raw_status, _ = await ops_test.juju(
+        "run-action", action_unit_name, "get-cluster-status", "--format=yaml", "--wait"
+    )
+
+    status = yaml.safe_load(raw_status.strip())
+
+    try:
+        cluster_topology = status[list(status.keys())[0]]["results"]["defaultreplicaset"][
+            "topology"
+        ]
+    except KeyError:
+        return False
+
+    for k, v in cluster_topology.items():
+        if k.replace("-", "/") == unit_name and v.get("status") == "online":
+            return True
+    raise TimeoutError
+
+
+def cut_network_from_unit(machine_name: str) -> None:
+    """Cut network from a lxc container.
+
+    Args:
+        machine_name: lxc container hostname
+    """
+    # apply a mask (device type `none`)
+    cut_network_command = f"lxc config device add {machine_name} eth0 none"
+    subprocess.check_call(cut_network_command.split())
+
+
+def restore_network_for_unit(machine_name: str) -> None:
+    """Restore network from a lxc container.
+
+    Args:
+        machine_name: lxc container hostname
+    """
+    # remove mask from eth0
+    restore_network_command = f"lxc config device remove {machine_name} eth0"
+    subprocess.check_call(restore_network_command.split())
+
+
+async def unit_hostname(ops_test: OpsTest, unit_name: str) -> str:
+    """Get hostname for a unit.
+
+    Args:
+        ops_test: The ops test object passed into every test case
+        unit_name: The name of the unit to be tested
+    Returns:
+        The machine/container hostname
+    """
+    _, raw_hostname, _ = await ops_test.juju("ssh", unit_name, "hostname")
+    return raw_hostname.strip()
+
+
+@retry(stop=stop_after_attempt(8), wait=wait_fixed(15))
+def wait_network_restore(model_name: str, hostname: str, old_ip: str) -> None:
+    """Wait until network is restored.
+
+    Args:
+        model_name: The name of the model
+        hostname: The name of the instance
+        old_ip: old registered IP address
+    """
+    if instance_ip(model_name, hostname) == old_ip:
+        raise Exception
+
+
+async def graceful_stop_server(ops_test: OpsTest, unit_name: str) -> None:
+    """Gracefully stop server.
+
+    Args:
+        ops_test: The ops test object passed into every test case
+        unit_name: The name of the unit to be tested
+    """
+    # send TERM signal to mysql daemon, which trigger shutdown process
+    await ops_test.juju("ssh", unit_name, "sudo", "pkill", "-15", "mysqld")
```

Comment:
I think you'll want to check the results of this to verify that kill was successful

Reply:
done